src/main/java/dev/memphis/sdk/MemphisConnection.java

```diff
@@ -0,0 +1,52 @@
+package dev.memphis.sdk;
+
+import dev.memphis.sdk.consumer.Consumer;
+import dev.memphis.sdk.producer.Producer;
+import io.nats.client.Connection;
+import io.nats.client.JetStream;
+import io.nats.client.Options;
+
+import java.util.concurrent.Future;
+
+public class MemphisConnection {
+
+    public MemphisConnection(Options natsConnOptions, Connection brokerConnection, JetStream jetStreamContext, String uuid) {
```

Comment:
We want to hide the underlying connection details.  I would make this method package private (remove the `public` keyword) so that the class can only be instantiated from within the Memphis connect() method.

Reply:
Alternatively, some of the connection setup can be moved into the constructor itself.  E.g., the constructor takes an options object.  This would also serve to encapsulate the underlying implementation details.